src/main/java/org/codehaus/gmavenplus/mojo/AbstractToolsMojo.java

```diff
@@ -70,15 +70,6 @@ public abstract class AbstractToolsMojo extends AbstractGroovyMojo {
     @Parameter
     protected Properties properties = new Properties();
 
-    /**
-     * Whether to allow System.exit() to be used. Should not be set to <code>false</code> when using parallel
-     * execution, as it isn't thread-safe.
-     *
-     * @since 1.2
-     */
-    @Parameter(defaultValue = "true")
-    protected boolean allowSystemExits;
-
```

Comment:
As this would be a breaking change, keep it for a while and just print a warning for now. This way, users can fix their scripts.

Reply:
Yea, that's why I hadn't merged this -- Java isn't forcing me to yet.

Another option here might be to wrap the usage in `if (Version.parseFromString(System.getProperty("java.version")).compareTo(new Version(24)) >=1)` and print a warning if using a newer Java version that won't support it and ignore the option in that case. And in the meantime, print a warning and deprecate the parameter.

Reply:
I'm not sure when they're planning on removing this (I used 24 as an example). I don't think they've decided when that will be (or at least I can't find any documentation about it).

Reply:
Yes, it gets removed. The warning is there since 17 or 18.

Docs: https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/SecurityManager.html

```
@Deprecated(since="17",
            forRemoval=true)
public class SecurityManager
extends Object
Deprecated, for removal: This API element is subject to removal in a future version.
The Security Manager is deprecated and subject to removal in a future release. There is no replacement for the Security Manager. See JEP 411 for discussion and alternatives.
```

There's more
https://openjdk.org/jeps/411

> Description

```
In Java 17, we will:

Deprecate, for removal, most Security Manager related classes and methods.

Issue a warning message at startup if the Security Manager is enabled on the command line.

Issue a warning message at run time if a Java application or library installs a Security Manager dynamically.

In Java 18, we will prevent a Java application or library from dynamically installing a Security Manager unless the end user has explicitly opted to allow it.
```

Reply:
> Another option here might be to wrap the usage in if (Version.parseFromString(System.getProperty(\"java.version\")).compareTo(new Version(24)) >=1)

But that would still print a ClassNotFoundException on 24 or so. Unless:

1. You use reflection. A lot.
2. Create a multi release jar
3. Drop it all together and print a warning

Reply:
I was thinking of using reflection

Reply:
I'd rather put that code into its own class.
Then, for Java 21+, create a Multi Release Jar which implements this as a no-op.

Easier to maintain, faster than reflection.

But still, why not start with a warning? 🤷🏻‍♂️ Ship early, ship often.